mgmt/src/mgmt.c

```diff
@@ -71,6 +71,31 @@ mgmt_streamer_free_buf(struct mgmt_streamer *streamer, void *buf)
     streamer->cfg->free_buf(buf, streamer->cb_arg);
 }
 
+void
+mgmt_unregister_group(struct mgmt_group *group)
+{
+    struct mgmt_group *curr = mgmt_group_list, *prev;
+
+    if (curr && curr == group) {
```

Comment:
There is no need to test `curr` here because if it is NULL it won't be equal to `group`

Reply:
Yes, agreed.

Reply:
The only reason I keep checking current everywhere is just to be safe since there was no NULL check for group.